src/lib/installation/update_repositories_finder.rb

```
@@ -124,7 +124,13 @@ def update_url_from_control
     #
     # @return [Array<URI>,false] self-update URLs or false in case of error
     def update_urls_from_connect
-      url = registration_url
+      begin
+        url = registration_url
+      rescue URI::InvalidURIError => e
+        log.error("Custom registration url is wrong, URI failed with: #{e.message}")
```

Yes, but silent fail should happen only in the default case. IMHO if the user uses the boot command line then an error should be printed.